design/clusters.go

```diff
@@ -106,21 +106,29 @@ var _ = a.Resource("clusters", func() {
 		a.Routing(
 			a.GET("/"),
 		)
+		a.Params(func() {
+			a.Param("type", d.String, "the type of the clusters to return ('OSD' or 'OSO')")
```

Comment:
You don't have to implement custom validation. You can use Enum() to specify allowed values of the param. Then goa will do payload validation for your.
See an example here - https://github.com/fabric8-services/fabric8-auth/blob/3a44ba7ddd1ebe96cae360c1cecfb7f7264d69d6/design/token.go#L124

Reply:
You mean JSON API Error response? It will. Check this out - https://auth.openshift.io/api/token/keys?format=unknown

Reply:
ok, good. Let me change that, then.

Reply:
@alexeykazakov addressed in https://github.com/fabric8-services/fabric8-cluster/pull/77/commits/cbc3cc727951770e7cf36944e435e9a34d4b9bbe and https://github.com/fabric8-services/fabric8-cluster/pull/77/commits/d85720dc335064076d4925b7c8cdf7412589b6b9